cmd/operator/api/v1/clusterkubearchiveconfig_webhook.go

```diff
@@ -115,6 +127,36 @@ func (ckaccv *ClusterKubeArchiveConfigCustomValidator) validateKAC(ckac *Cluster
 				errList = append(errList, validateDurationString(resource.ArchiveOnDelete)...)
 			}
 		}
+
+		// Validate KeepLastWhen rules
+		seenCELExpressions := make(map[string]string)
+		for i, rule := range resource.KeepLastWhen {
+			if rule.Name == "" {
+				errList = append(errList, fmt.Errorf("keepLastWhen[%d].name is required", i))
+			}
+			if rule.When == "" {
+				errList = append(errList, fmt.Errorf("keepLastWhen[%d].when is required", i))
+			} else {
+				_, err := cel.CompileCELExpr(rule.When)
+				if err != nil {
+					errList = append(errList, fmt.Errorf("keepLastWhen[%d].when: %w", i, err))
+				} else {
+					durErrors := validateDurationString(rule.When)
+					for _, durErr := range durErrors {
+						errList = append(errList, fmt.Errorf("keepLastWhen[%d].when: %w", i, durErr))
+					}
+				}
+
+				if existingRuleName, exists := seenCELExpressions[rule.When]; exists {
+					errList = append(errList, fmt.Errorf("keepLastWhen[%d].when CEL expression duplicates rule '%s'; duplicate expressions are not allowed", i, existingRuleName))
```

Comment:
Here we validate if the CEL expression is duplicated, but I think we should also check if the name is duplicated

Reply:
Fixed.